internal/services/synapse/synapse_pipeline_resource_test.go

```diff
@@ -0,0 +1,367 @@
+	}
+
+	environment := client.Account.Environment
+	pipelinesClient, err := client.Synapse.PipelinesClient(id.WorkspaceName, environment.SynapseEndpointSuffix)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := pipelinesClient.GetPipeline(ctx, id.Name, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return utils.Bool(false), nil
+		}
+		return nil, fmt.Errorf("retrieving retrieving %s: %+v", id, err)
+	}
+
+	return utils.Bool(true), nil
+}
+
+func (r PipelineResource) basic(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+	%s
+
+resource "azurerm_synapse_pipeline" "test" {
+  name                 = "acctest-SynapsePipeline%d"
+  synapse_workspace_id = azurerm_synapse_workspace.test.id
+
+  depends_on = [azurerm_synapse_firewall_rule.test]
+}
+`, template, data.RandomInteger)
+}
+
+func (r PipelineResource) complete(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+	%s
+
+resource "azurerm_synapse_pipeline" "test" {
+  name                 = "acctest-SynapsePipeline%d"
+  synapse_workspace_id = azurerm_synapse_workspace.test.id
+
+  parameters = {
+    test = "testparameter"
+  }
+
+  variables = {
+    foo = "test1"
+    bar = "test2"
+  }
+
+  depends_on = [azurerm_synapse_firewall_rule.test]
+}
+`, template, data.RandomInteger)
+}
+
+func (r PipelineResource) update1(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+	%s
+
+resource "azurerm_synapse_pipeline" "test" {
+  name                 = "acctest-SynapsePipeline%d"
+  synapse_workspace_id = azurerm_synapse_workspace.test.id
+  annotations          = ["test1", "test2", "test3"]
+  description          = "test description"
+
+  parameters = {
+    test = "testparameter"
+  }
+
+  variables = {
+    foo = "test1"
+    bar = "test2"
+  }
+
+  depends_on = [azurerm_synapse_firewall_rule.test]
+}
+`, template, data.RandomInteger)
+}
+
+func (r PipelineResource) update2(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+	%s
+
+resource "azurerm_synapse_pipeline" "test" {
+  name                 = "acctest-SynapsePipeline%d"
+  synapse_workspace_id = azurerm_synapse_workspace.test.id
+  annotations          = ["test1", "test2"]
+  description          = "test description2"
+
+  parameters = {
+    test  = "testparameter"
+    test2 = "testparameter2"
+  }
+
+  variables = {
+    foo = "test1"
+    bar = "test2"
+    baz = "test3"
+  }
+
+  activities_json = <<JSON
+[
+  {
+    "name": "Append variable1",
+    "type": "AppendVariable",
+    "dependsOn": [],
+    "userProperties": [],
+    "typeProperties": {
+      "variableName": "foo",
+      "value": "something"
+    }
+  },
+  {
+    "name": "Append variable2",
+    "type": "AppendVariable",
+    "dependsOn": [],
+    "userProperties": [],
+    "typeProperties": {
+      "variableName": "bar",
+      "value": "something1"
+    }
+  }
+]
+JSON
```

Comment:
Why is this json and not a block in schema? it seems like it is a list of blocks - we should do this regardless of what the datafactory resource is doing, and then at some point go back and fix the data factory resource

Reply:
It is hard to do so. There are ~40 different activity types. Can we keep it as it is now or change it to a string list?

Reply:
i'm not sure i follow, if we change this to an activity block it should be
```suggestion
  activities = 
[
  {
    name = "Append variable1"
    type = "AppendVariable"
    dependsOn =  []
    user_uroperties = []
    type_properties {
      "name": "foo",
      "value": "something"
    }
 
```

and so forth

Reply:
@katbyte each type has its own set of properties, i.e. a variant of the discriminator. This is the same as the data factory pipeline's `activities_json`. I'd suggest to make it a `list` of `string`, where each `string` is a json. WDYT?

Reply:
could you demonstrate what you mean/have a different type here so it's clear the differences because right now there is only append variables used

Reply:
Of course. [Here](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json#L6991) is the Swagger definition of the `AppendVariable` type. Similarly, if you search for `discriminator-value` in the Swagger, there are 37 matches where each represents a variant of the `Activity` discriminator.

Also, [here](https://github.com/hashicorp/terraform-provider-azurerm/blob/291780012c30d974c87c1ed3d7f9a16257a02c84/internal/services/datafactory/data_factory_pipeline_resource.go#L75) is how the data factory handles this.

Reply:
@katbyte , can we proceed with list of `string` way as @magodo suggests?

Reply:
Yep, we discussed internally and while not ideal its all we can do for now